src/lumigo_tracer/lumigo_utils.py

```diff
@@ -547,34 +553,51 @@ def _recursive_omitting(
     d, free_space = prev_result
     if free_space < 0:
         return d, free_space
+    should_skip_key = False
+    if isinstance(d, dict) and omit_skip_path:
+        should_skip_key = omit_skip_path[0] == key
+        omit_skip_path = omit_skip_path[1:] if should_skip_key else None
     if key in SKIP_SCRUBBING_KEYS:
-        d[key] = value
-        free_space -= len(value) if isinstance(value, str) else len(aws_dump(d))
-    elif isinstance(key, str) and regex and regex.match(key):
-        d[key] = "****"
+        new_value = value
+        free_space -= len(value) if isinstance(value, str) else len(aws_dump({key: value}))
```

Comment:
```suggestion
        free_space -= len(value) if isinstance(value, str) else len(aws_dump(d))
```
If `d` was list then this is wierd